### PR TITLE
Freeradius configuration files parser (/etc/raddb/*)

### PR DESCRIPTION
**Pull request by rmetrich**
See Master PR https://github.com/RedHatInsights/insights-core/pull/841

### CODE REVIEW

File: insights/parsers/freeradius.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+FreeRadius - Freeradius configuration files
+===========================================
+
+FreeRadius is a parser for the freeradius configuration files usually in
+``/etc/raddb``.
+Files ending with ``.rpmnew`` are excluded because they are backups made when
+ugprading but customer modified the configuration files, so are not used by the
+service.
+
+Because this parser reads multiple files, the files are stored as a list within
```
Comment (PaulWay) [inline insights/parsers/freeradius.py]: As a side note, this is a perfect candidate for making a combiner that combines all the configuration into one cohesive whole.  See the limits.d and modprobe combiners for examples.

File: insights/parsers/freeradius.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+the parser and need to be iterated through in order to find specific
+configuration files.
+
```
Comment (PaulWay) [inline insights/parsers/freeradius.py]: Really needs examples of what the files look like and a discussion of the format they're going to be stored in.

File: insights/parsers/freeradius.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+Examples:
+
+    >>> for cfgfile in shared[FreeRadius]: # Parser contains list of all configuration files
+    ...     print "Path:", cfgfile.file_path
+    ...
+    Path: etc/raddb/radiusd.conf
+    Path: etc/raddb/proxy.conf
+    Path: etc/raddb/policy.d/filter
+
```
Comment (PaulWay) [inline insights/parsers/freeradius.py]: Really _REALLY_ needs better examples, mainly detailing what is in each `cfgfile` and how to get configuration information.

File: insights/parsers/freeradius.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        # Ignore ".rpmnew" files, these are backups made when upgrading but
+        # customer modified the configuration files
+        if self.file_name.endswith('.rpmnew'):
+            raise SkipException("Skipping '.rpmnew' file.")
+        self.file_content = content
+        self.file_content_stripped = []
+        for l in content:
+            s = l.strip()
+            if s and not s.startswith('#'):
+                self.file_content_stripped.append(s)
```
Comment (PaulWay) [inline insights/parsers/freeradius.py]: Personally I would like this to do a lot more.  There are many other good examples of configuration file parsers that store structured information.  Key/value pairs as a dictionary?  Structured information?  Go for it!

And I'm really not sure what the point of storing the whole content _as well as_ only the active lines.  What are you intending to do with it?

File: insights/parsers/freeradius.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    """
+
+    def parse_content(self, content):
+        # Ignore ".rpmnew" files, these are backups made when upgrading but
+        # customer modified the configuration files
+        if self.file_name.endswith('.rpmnew'):
+            raise SkipException("Skipping '.rpmnew' file.")
+        self.file_content = content
+        self.file_content_stripped = []
+        for l in content:
```
Comment (PaulWay) [inline insights/parsers/freeradius.py]: This should really use the `get_active_lines` function to get only lines containing content that doesn't start with '#'.

File: insights/parsers/tests/test_freeradius.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
+import pytest
+
+from insights.parsers import SkipException
+from insights.parsers.freeradius import FreeRadius
+from insights.tests import context_wrap
+
+RADIUSD_CONF_FILE = "etc/raddb/radiusd.conf"
+RADIUSD_CONF_CONTENT = """
```
Comment (PaulWay) [inline insights/parsers/tests/test_freeradius.py]: Consider using these files as samples in your documentation.

File: insights/parsers/freeradius.py (hunk 3 above)
Comment (PaulWay) [reply insights/parsers/freeradius.py]: Then you can test it using the `doctest` framework - see the examples in http://insights-core.readthedocs.io/en/1.x/docs_guidelines.html.  I would recommend also using the doctest-as-test system - I've got https://github.com/RedHatInsights/insights-core/pull/845 as a merge to improve the documentation update but examples can be found in #817.

That way you the examples can also be your tests!

File: insights/parsers/tests/test_freeradius.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+localstatedir = /var
+sbindir = /usr/sbin
+
+log {
+        #  The command-line option "-X" over-rides this option, and forces
+        #  logging to go to stdout.
+        #
+        destination = files
+}
+""".strip()
```
Comment (PaulWay) [inline insights/parsers/tests/test_freeradius.py]: There's no point in stripping this string if it's going to be handled as line content that is stripped anyway.

File: insights/parsers/freeradius.py (hunk 4 above)
Comment (PaulWay) [reply insights/parsers/freeradius.py]: IOW _DO NOT_ leave the rule writer to implement your parsing for you.

**Comment by eduardocerqueira** [comment]
Can one of the admins verify this patch?

**Comment by bfahr** [comment]
ok to test

**Comment by xiangce** [comment]
Since the same PR to the master branch was already closed and the 1.x branch will soon be deprecated. I go ahead and close this PR. Please feel free to open another PR to the master branch to re-submit it if this parser is needed by your rules. Thanks.

**Pull request closed without merging**